sympy/physics/mechanics/beam.py

```diff
@@ -0,0 +1,263 @@
+"""
+This module can be used to solve beam bending problems in mechanics.
+
+"""
+
+from __future__ import print_function, division
+
+from sympy.printing import sstr
+from sympy.physics.mechanics import Point
+from sympy import sympify
+
+
+class Beam(object):
+
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending.
+    Beams are characterized by their profile (moment of Inertia), their length,
+    and their material.
+
+    Parameters
+    ==========
+    length : Sympifyable
+        A SymPy expression representing the Beam's length.
+    E : Sympifyable
+        A SymPy expression representing the Beam's Young's Modulus.
+        It is a measure of the stiffness of the Beam material.
+    I : Sympifyable
+        A SymPy expression representing the Beam's Moment of Inertia.
+        It determines the torque needed for a desired angular acceleration
+        about a rotational axis of the Beam.
+
+    Examples
+    ========
+    >>> from sympy.physics.mechanics.beam import Beam
+    >>> from sympy import Symbol
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> Beam(1, E, I)
+    Beam(1, E, I)
+
+    """
+
+    def __init__(self, length, E, I):
```

Comment:
I think only  "length", "elastic_modulus", and "second_moment" should be used to initialize the beam object. The others should be supplied later. This would make the API clean as well as more handy for beginners.